test/dhtest.c

```diff
@@ -535,10 +543,20 @@ static int run_rfc5114_tests(void)
     }
     return 1;
  bad_err:
+    DH_free(dhA);
```

Comment:
aha, sorry for that mistake, I think the latest commit has fix the scope issue, thank you all:)